src/app/form-entry/control-group.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { FormControl, FormGroup, Validators, FormBuilder, FormArray } from '@angular/forms';
+import { Validation } from './models/validation';
+import { QuestionBase } from './models/question-base';
+import { QuestionGroup } from './models/question-group';
+@Injectable()
+export class ControlGroupService {

```

Comment:
Consider renaming to 'FormControlService' or 'FormControFactory'.
This service is used to 'assemble' or 'manufacture' angular form controls inheriting AbstractControl. Renaming it to a factory promotes adoption of the factory pattern.